ios/CustomerioReactnative.swift

```diff
@@ -208,6 +209,26 @@ class CustomerioReactnative: NSObject {
         })
     }
     
+    // Tracks `opened` push metrics when a push notification is interacted with.
+    @objc(trackNotificationResponseReceived:)
+    func trackNotificationResponseReceived(payload: NSDictionary) {
+        trackPushMetrics(payload: payload, event: .opened)
+    }
+    
+    // Tracks `delivered` push metrics when a push notification is received.
+    @objc(trackNotificationReceived:)
+    func trackNotificationReceived(payload: NSDictionary) {
+        
+        trackPushMetrics(payload: payload, event: .delivered)
+    }
+    
+    private func trackPushMetrics(payload: NSDictionary, event : Metric) {
+        guard let deliveryId = payload["CIO-Delivery-ID"] as? String, let deviceToken = payload["CIO-Delivery-Token"] as? String else
+        {return}
+        
+        MessagingPush.shared.trackMetric(deliveryID: deliveryId, event: event, deviceToken: deviceToken)
```

Comment:
Just curious, why do we need exactly same implementation in both `ios/CustomerioPushMessaging.swift` and `ios/CustomerioReactnative.swift`? 🤔

Reply:
Thanks for the highlight. This is now fixed.